spec/requests/posts_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+describe 'GET /index' do
+  it 'returns a successful response' do
+    user = User.create!(name: 'User1')
+    get "/users/#{user.id}/posts/index"
+    expect(response).to be_successful
+  end
```

Comment:
![image](https://github.com/Nessrine88/Blog-App/assets/78251927/4f77d5af-4ea3-4f00-b025-276a451458ee)

The issue at hand is that the error message "Posts counter is not a number" indicates that in order to create a new `user`, our validation process necessitates the presence of an initialized integer value for the `posts_counter` field. 

We can address this by ensuring that `posts_counter` is appropriately initialized as an integer. 🙂💪

```sql
user = User.create!(name: 'User1', posts_counter: 0)
```